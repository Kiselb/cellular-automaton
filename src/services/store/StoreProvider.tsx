import React, { ReactNode } from "react";
import { Provider } from "react-redux";

import { AppStoreMode } from "../../domain/types";
import storeThunk from "../../store/ducks/store";
import storeSaga from "../../store/sagas/index";

type StoreProviderProps = {
  mode: AppStoreMode;
  children: ReactNode;
};
export const StoreProvider = ({ mode, children }: StoreProviderProps) => {
  return mode === "ReduxThunk" || mode === "ReduxSaga" ? (
    mode === "ReduxThunk" ? (
      <Provider store={storeThunk}>{children}</Provider>
    ) : (
      <Provider store={storeSaga}>{children}</Provider>
    )
  ) : (
    <>{children}</>
  );
};
