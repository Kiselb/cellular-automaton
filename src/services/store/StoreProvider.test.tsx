import React from "react";
import { render, screen } from "@testing-library/react";

import { StoreProvider } from "./StoreProvider";

it("App renders correctly", () => {
  localStorage.setItem("cellular-automaton.user", "Native");
  const { asFragment } = render(
    <StoreProvider mode={"Native"}>Native</StoreProvider>
  );
  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
it("App renders correctly", () => {
  localStorage.setItem("cellular-automaton.user", "ReduxThunk");
  const { asFragment } = render(
    <StoreProvider mode={"ReduxThunk"}>ReduxThunk</StoreProvider>
  );
  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
it("App renders correctly", () => {
  localStorage.setItem("cellular-automaton.user", "ReduxSaga");
  const { asFragment } = render(
    <StoreProvider mode={"ReduxSaga"}>ReduxSaga</StoreProvider>
  );
  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
it("App renders correctly", () => {
  localStorage.setItem("cellular-automaton.user", "ReduxEffects");
  const { asFragment } = render(
    <StoreProvider mode={"ReduxEffects"}>ReduxEffects</StoreProvider>
  );
  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
