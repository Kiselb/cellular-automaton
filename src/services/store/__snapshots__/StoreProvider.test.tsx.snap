// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  Native
</DocumentFragment>
`;

exports[`App renders correctly 2`] = `
<DocumentFragment>
  ReduxThunk
</DocumentFragment>
`;

exports[`App renders correctly 3`] = `
<DocumentFragment>
  ReduxSaga
</DocumentFragment>
`;

exports[`App renders correctly 4`] = `
<DocumentFragment>
  ReduxEffects
</DocumentFragment>
`;
